rhttpc-client/src/main/scala/rhttpc/client/proxy/FailureResponseHandleStrategyChooser.scala

```diff
@@ -45,18 +45,25 @@ case class BackoffRetry(initialDelay: FiniteDuration,
   override def choose(currentRetryAttempt: Int,
                       lastPlannedDelay: Option[FiniteDuration],
                       dateOfFirstAttempt: Instant): ResponseHandleStrategy = {
+    val nextDelay = lastPlannedDelay match {
+      case Some(lastDelay) =>
+        (lastDelay.toMillis * multiplier).toLong
+      case None =>
+        initialDelay.toMillis
+    }
+
     deadline match {
       case Some(duration) =>
         val jDuration = java.time.Duration.ofMillis(duration.toMillis)
         val deadlineDate = dateOfFirstAttempt.plus(jDuration)
 
-        if (Instant.now().isBefore(deadlineDate)) {
-          BackoffRetries.chooseBasedOnRetries(currentRetryAttempt, lastPlannedDelay, initialDelay, maxRetries, multiplier)
+        if (Instant.now().plusMillis(nextDelay).isBefore(deadlineDate)) {
```

Comment:
During previous review I've missed the fact that u inlined `Instant.now()` here. Can u extract it to method parameter `now`? And use `timeProvider` in place of invocation? If you had written the test for that you would have seen that :-P Maybe it is a good time to wrote tests for let's say two cases?